src/hotspot/share/compiler/compilerThread.hpp

```diff
@@ -38,10 +45,25 @@ class CompileQueue;
 class CompilerCounters;
 class IdealGraphPrinter;
 
+class CompilerThreadTimeoutGeneric : public CHeapObj<mtCompiler> {
+ public:
+  CompilerThreadTimeoutGeneric() {};
+  void arm() { return; };
+  void disarm() { return; };
+  bool init_timeout() { return true; };
+};
```

Comment:
Should we also guard this with `ifndef LINUX` since it's only used for non-Linux?

Reply:
Sounds reasonable.